src/KafkaFlow/Extensions/ConfigurationBuilderExtensions.cs

```diff
@@ -120,5 +124,90 @@ public static IConsumerConfigurationBuilder WithConsumerConfig(this IConsumerCon
         {
             return ((ProducerConfigurationBuilder) builder).WithCustomFactory(decoratorFactory);
         }
+
+        /// <summary>
+        /// Adds typed handler middleware
+        /// </summary>
+        /// <param name="builder">Instance of <see cref="IConsumerMiddlewareConfigurationBuilder"/></param>
+        /// <param name="configure">A handler to configure the middleware</param>
+        /// <returns></returns>
+        public static IConsumerMiddlewareConfigurationBuilder AddTypedHandlers(
+            this IConsumerMiddlewareConfigurationBuilder builder,
+            Action<TypedHandlerConfigurationBuilder> configure)
+        {
+            var typedHandlerBuilder = new TypedHandlerConfigurationBuilder(builder.DependencyConfigurator);
+
+            configure(typedHandlerBuilder);
+
+            var configuration = typedHandlerBuilder.Build();
+
+            builder.Add(
+                resolver => new TypedHandlerMiddleware(resolver, configuration),
+                MiddlewareLifetime.Scoped);
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Registers a middleware to decompress the message
+        /// </summary>
+        /// <param name="middlewares">The middleware configuration builder</param>
+        /// <typeparam name="T">The compressor type</typeparam>
+        /// <returns></returns>
+        [Obsolete("Compressors should only be used in backward compatibility scenarios, in the vast majority of cases native compression (producer.WithCompression()) should be used instead")]
```

Comment:
The obsoletes are to be removed as discussed here: https://github.com/Farfetch/kafkaflow/pull/409#discussion_r1338304475

(The same for lines 172, 188 and 204)

Reply:
As discussed with @filipeesch these methods will be kept for now